driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import org.bson.BsonBoolean;
+import org.bson.BsonDateTime;
+import org.bson.BsonDecimal128;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonInt64;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.conversions.Bson;
+import org.bson.types.Decimal128;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+
+/**
+ * Builders for aggregate expressions.
+ *
+ * @mongodb.driver.manual reference/operator/aggregation/
+ * @since 4.?
+ */
+public interface Expressions {
+
+    //
+    //  Literal expressions
+    //
+
+    // The regular literal expressions will _not_ wrap the values in $literal. This is the normal case so making these
+    // method names shorter ones
+
+    /**
+     * An expression representing the given boolean value.
+     *
+     * @param value the boolean value of the expression
+     * @return a literal expression representing the given boolean value
+     * @mongodb.driver.manual meta/aggregation-quick-reference/#literals
+     */
+    static LiteralExpression literal(final boolean value) {
+        return new LiteralExpression(BsonBoolean.valueOf(value), true);
+    }
+
+    static LiteralExpression literal(final int value) {
+        return new LiteralExpression(new BsonInt32(value), true);
+    }
+
+    static LiteralExpression literal(final long value) {
+        return new LiteralExpression(new BsonInt64(value), true);
+    }
+
+    static LiteralExpression literal(final Date value) {
```

Comment:
Maybe it's stupid to have these overloads, which can create ambiguity with the one below that takes Object.  It means that if you want a null literal you need to write `literal((Object) null)`.  Alternatively could create a special method for null literals, e.g. `nullLiteral()`.

Reply:
I think nullLiteral works, I'd rather be type friendly and explicit.